distribution/docker/build.gradle

```diff
@@ -170,11 +192,14 @@ void addBuildDockerImage(final boolean oss) {
     }
   }
   assemble.dependsOn(buildDockerImageTask)
+  buildDockerImageTask.enabled = Architecture.current().name().toLowerCase().equals(architecture)
```

Comment:
Let's standardize on using `onlyIf { }` for this kind of thing going forward.

Reply:
I opened #54047.